### PR TITLE
[Feature] First Nations status radio group

### LINKED ISSUE #7525: ♻️ Rework the First Nations identity interaction in the candidate DEI dialog and the Self-declaration form
## ♻️ Summary

As a part of the decision to not use related checkboxes for single-selection of the two options, the decision is to reverse the change and return to a single checkbox that then triggers a radio set. As a part of these changes, the interaction has been updated to provide a clearer interaction for both the DEI profile dialog and the Self-declaration component.

Resolves:
- #5388
- #7360

## 🗨️ DEI profile dialog

[View the Figma file here](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=288-42987&mode=design&t=zVXKevgu3VuWAP15-4) | Please contact @substrae with questions on this portion of the ticket.

```[tasklist]
### Acceptance Criteria
- [ ] The "First Nations" checkbox label now identifies status to help clarify the intent of the option
- [ ] The signature copy and input is now hidden if all 4 options are unselected
- [ ] Whitespace has been updated to help with breathing room
- [ ] Identity icons have been realigned with their respective inputs to ensure the animal's feet generally align with the bottom of the input
- [ ] Criteria met in the DEI dialog and the application self-declaration step
``` 

image

## 👤 Self-declaration component

Please contact @BrookeCameron with questions on this portion of the ticket.

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7525 

## 👋 Introduction

Updates the Indigenous self-declaration form to use a radio group for First Nations status.

## 🕵️ Details

This was done to force a user to only select one or the other but not both. Some other changes were made as well:

- Spacing has been updated
- Signature field hidden when no community is selected
- Submit button changes content based on the form values (removes the term "sign" when no signature field appears)

> **Note**
> I left in the "select both" error message since we already have it and to guard against any edge cases

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Navigate to your profile
3. Scroll down to the DEI section
4. Open the Indigenous identify dialog
5. Interact with the form and confirm the First Nations status is required when selecting that as a community
6. Confirm values are submitted to the API properly

## 📸 Screenshot

Screenshot 2023-10-13 092543

### CODE REVIEW

File: apps/web/src/components/EmploymentEquity/dialogs/IndigenousDialog.tsx
Comment (vd1992) [inline apps/web/src/components/EmploymentEquity/dialogs/IndigenousDialog.tsx:1]: Some layout things, but probably a separate issue since they come from a separate component right?

Text can overlap the bolded text on the warn dialog on smaller viewports. 

image

And the bolded text also is a bit squished on larger viewports.

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Looks good, some things that are more questions than hard requests, please take a look

File: apps/web/src/components/EmploymentEquity/dialogs/IndigenousDialog.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/EmploymentEquity/dialogs/IndigenousDialog.tsx:1]: That is entirely intentional. The alert is because they selected a community while at the same time saying their community is in the list. It is a warning incase they didn't see their community at first and selected it.

File: apps/web/src/utils/indigenousDeclaration.ts
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/utils/indigenousDeclaration.ts:1]: That makes sense to me! Updated in [null signature when communities are empty](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8224/commits/c7667f4607eefb08ecd9c190ed5de422e204fa0b)

[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/utils/indigenousDeclaration.ts:1]: Hm, it appears to be a bit odd at the moment. It doesn't save things correctly all the time. 

Please see the recording, sorry for the 10 seconds of confusion I had with network log tabs 
https://recordit.co/ly4IDe0a15

[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/utils/indigenousDeclaration.ts:1]: Good catch, looks like I selected the wrong variable in autosuggest 😅 

Fixed in [fix typo in community length check for signature](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8224/commits/dfac5db37d352e4d037e739c040c56c114769761)

[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/utils/indigenousDeclaration.ts:1]: There we go, it is working

**Review by vd1992 — APPROVED** [review]
🎖️ 

image

image

File: apps/web/src/lang/fr.json [hunk 1]
```diff
@@ -5382,6 +5386,10 @@
     "defaultMessage": "Le droit de priorité est un statut offert par la Commission de la fonction publique du Canada. Pour en apprendre davantage, <priorityEntitlementLink>consultez le site Web Information sur les droits de priorité</priorityEntitlementLink>.",
     "description": "Sentence describing what priority entitlement is"
   },
+  "Wpj2OY": {
+    "defaultMessage": "\"Je suis membre d’une Première Nation\"",
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:5390]: @esizer Shouldn't this use _guillemets_ instead of quotes?

`"defaultMessage": "« Je suis membre d’une Première Nation »",`

Comment (esizer) [reply apps/web/src/lang/fr.json:5390]: Oh probably, that is just how I received it from translation services 🤷‍♀️
